react-app/src/utils/leaguehelpers.js

```diff
@@ -33,51 +33,115 @@ export default {
 
 	getDate: (league, dateId) => {
 		//dateId: league.scheduledMatches.dateId
-		if(league == null || dateId === ''){
+		var getdate = {};
+
+		if(league === null || dateId === ''){
 			return null;
 		}
 
 		league.dates.forEach((date) => {
 			if(date.id === dateId){
-				return date;
+				getdate = date;
 			}
 		});
+
+		//return leagues.dates.filter((date) => {date.id === dateId});
+
+		return getdate;
 	},
 
 	getNumInLeague: (league, teamId) => {
 		//teamId: league.scheduledMatches.teamOneId
-		if(league === null || dateId === ''){
+		var teamNum = '';
+
+		if(league === null || teamId === ''){
 			return '';
 		}
 
 		league.teams.forEach((team) => {
 			if(team.id === teamId){
-				return team.numInLeague;
+				teamNum = team.numInLeague;
 			}
 		});
+		
+
+		//return leagues.teams.filter((team) => {team.id === teamId});
```

Comment:
Can you switch these util functions to use filters instead of forEach's

Reply:
filter will return an array with 1 date object inside, not sure why I would want to do this as oppose to just having a single date object that I can access